models/models.js

```diff
@@ -7,6 +7,29 @@ exports.selectCategories = () => {
         })
 }
 
+exports.selectReviews = (query) => {
+    const queryValue = [];
+    let conditionStr = ""
+
+    if (query.category) {
+        conditionStr += ` WHERE category = $1`
+        queryValue.push(query.category)
+    }
+
+    let queryStr = `SELECT reviews.*, COUNT(comments.review_id)::INT AS comment_count
+    FROM reviews 
+    LEFT JOIN comments ON reviews.review_id = comments.review_id
+    ${conditionStr}
+    GROUP BY reviews.review_id`
+
+    return db.query(queryStr, queryValue)
+        .then(({rows}) => {
+            const reviews = rows
+            if (reviews.length < 1) return Promise.reject({status: 400, msg: 'Bad Request'})
+            return reviews
```

Comment:
the situation I mentioned:
-here if reviews.length is 0 - this could actually mean that there is a category but is has no review yet
-in that case we need to find out if the category exists (200) or if the category doesn't exist (400)